docs/CustomApp.md

```diff
@@ -19,22 +19,44 @@ Here is the default store creation for react-admin:
 
 ```js
 // in src/createAdminStore.js
-import { applyMiddleware, combineReducers, compose, createStore } from 'redux';
-import { routerMiddleware, connectRouter } from 'connected-react-router';
+import { createStore, compose, applyMiddleware } from 'redux';
+import { routerMiddleware } from 'connected-react-router';
 import createSagaMiddleware from 'redux-saga';
 import { all, fork } from 'redux-saga/effects';
-import {
-    adminReducer,
-    adminSaga,
-    defaultI18nProvider,
-    USER_LOGOUT,
-} from 'react-admin';
+import { History } from 'history';
+
+import { AuthProvider, DataProvider, I18nProvider } from './types';
+import createAppReducer from './reducer';
+import { adminSaga } from './sideEffect';
+import { CLEAR_STATE } from './actions/clearActions';
+
+interface Window {
+    __REDUX_DEVTOOLS_EXTENSION_COMPOSE__?: (traceOptions: object) => Function;
+}
+
+export type InitialState = object | (() => object);
+
+interface Params {
+    dataProvider: DataProvider;
+    history: History;
+    authProvider?: AuthProvider;
+    customReducers?: any;
+    customSagas?: any[];
+    i18nProvider?: I18nProvider;
+    initialState?: InitialState;
+    locale?: string;
+}
```

Comment:
The documentation is in JavaScript, not TypeScript

Reply:
Ok, I think I got it right now.